### PR TITLE
Migrate remaining logs to structured logging

### PR DESCRIPTION
**Pull request by ehashman**
#### What type of PR is this?

/kind cleanup

/sig node
/sig instrumentation
/priority important-soon
/milestone v1.21

#### What this PR does / why we need it:

Migrates files to structured logging in unresponsive PRs #97569, #97520, #98632.

#### Which issue(s) this PR fixes:

Related: https://github.com/kubernetes/kubernetes/issues/98976

#### Does this PR introduce a user-facing change?

```release-note
kubelet: completed support for structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: https://github.com/kubernetes/enhancements/issues/1602
```

### CODE REVIEW

**Review by kikisdeliveryservice — COMMENTED** [review]
full update to structured logging for these files :+1: 

/lgtm

File: pkg/kubelet/kuberuntime/kuberuntime_gc.go [hunk 1]
```diff
@@ -342,7 +342,7 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 			}
 			err := osInterface.RemoveAll(filepath.Join(podLogsRootDirectory, name))
 			if err != nil {
-				klog.Errorf("Failed to remove pod logs directory %q: %v", name, err)
+				klog.ErrorS(err, "Failed to remove pod logs directory", "path", name)
```
Comment (navidshaikh) [inline pkg/kubelet/kuberuntime/kuberuntime_gc.go:345]:
```suggestion
				klog.ErrorS(err, "Failed to remove pod logs directory", "path", filepath.Join(podLogsRootDirectory, name))
```

Comment (navidshaikh) [reply pkg/kubelet/kuberuntime/kuberuntime_gc.go:345]: not required to change given the scope of the PR, but just a suggestion to log abs path

**Review by navidshaikh — COMMENTED** [review]
lgtm

File: pkg/kubelet/kuberuntime/kuberuntime_gc.go [hunk 2]
```diff
@@ -370,17 +370,17 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
...
-				klog.V(4).Infof("unable to obtain container Id: %v", err)
+				klog.V(4).InfoS("Unable to obtain container ID", "err", err)
 			}
 			err := osInterface.Remove(logSymlink)
 			if err != nil {
-				klog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+				klog.ErrorS(err, "Failed to remove container log dead symlink", "path", logSymlink)
 			} else {
-				klog.V(4).Infof("removed symlink %s", logSymlink)
+				klog.V(4).Infof("Removed symlink", "path", logSymlink)
```
Comment (adisky) [inline pkg/kubelet/kuberuntime/kuberuntime_gc.go:383]:
```suggestion
				klog.V(4).InfoS("Removed symlink", "path", logSymlink)
```

Comment (ehashman) [reply pkg/kubelet/kuberuntime/kuberuntime_gc.go:383]: Oops, will rebase :)

File: pkg/kubelet/kuberuntime/kuberuntime_gc.go (hunk 1 above)
Comment (ehashman) [reply pkg/kubelet/kuberuntime/kuberuntime_gc.go:345]: It's possible the `err` is due to that call failing? So I don't know that we should do that, but we could add a keypair with the root directory in the future.

**Comment by ehashman** [comment]
@kikisdeliveryservice PTAL

**Comment by kikisdeliveryservice** [comment]
Looks like: https://github.com/kubernetes/kubernetes/issues/95600

/test pull-kubernetes-e2e-kind

**Comment by kikisdeliveryservice** [comment]
seems like maybe: https://github.com/kubernetes/kubernetes/issues/99979

/retest

**Pull request merged**